upup/pkg/fi/cloudup/awstasks/classic_load_balancer.go

```diff
@@ -80,6 +84,114 @@ func (e *ClassicLoadBalancer) CompareWithID() *string {
 	return e.Name
 }
 
+type deleteTargetGroup struct {
+	request *elbv2.DeleteTargetGroupInput
+}
+
+var _ fi.Deletion = &deleteTargetGroup{}
+
+func (d *deleteTargetGroup) TaskName() string {
+	return "TargetGroup"
+}
+
+func (d *deleteTargetGroup) Item() string {
+	return aws.StringValue(d.request.TargetGroupArn)
+}
+
+func (d *deleteTargetGroup) Delete(t fi.Target) error {
+	klog.V(2).Infof("deleting target group %v", d)
+
+	awsTarget, ok := t.(*awsup.AWSAPITarget)
+	if !ok {
+		return fmt.Errorf("unexpected target type for deletion: %T", t)
+	}
+
+	klog.V(2).Infof("Calling Nlb DeleteTargetGroup for %s", aws.StringValue(d.request.TargetGroupArn))
+
+	if _, err := awsTarget.Cloud.ELBV2().DeleteTargetGroup(d.request); err != nil {
+		return fmt.Errorf("error Deleting TargetGroup from NLB: %v", err)
+	}
+
+	return nil
+}
+
+func (d *deleteTargetGroup) String() string {
+	return d.TaskName() + "-" + d.Item()
+}
+
+type deleteNetworkLoadBalancer struct {
+	request *elbv2.DeleteLoadBalancerInput
+}
+
+var _ fi.Deletion = &deleteNetworkLoadBalancer{}
+
+func (d *deleteNetworkLoadBalancer) TaskName() string {
+	return "NetworkLoadBalancer"
+}
+
+func (d *deleteNetworkLoadBalancer) Item() string {
+	return aws.StringValue(d.request.LoadBalancerArn)
+}
+
+func (d *deleteNetworkLoadBalancer) Delete(t fi.Target) error {
+	klog.V(2).Infof("deleting nlb %v", d)
+
+	awsTarget, ok := t.(*awsup.AWSAPITarget)
+	if !ok {
+		return fmt.Errorf("unexpected target type for deletion: %T", t)
+	}
+
+	klog.V(2).Infof("Calling elb DeleteLoadBalancer for %s", aws.StringValue(d.request.LoadBalancerArn))
+
+	if _, err := awsTarget.Cloud.ELBV2().DeleteLoadBalancer(d.request); err != nil {
+		return fmt.Errorf("error deleting nlb %s: %v", aws.StringValue(d.request.LoadBalancerArn), err)
+	}
+
+	return nil
+}
+
+func (d *deleteNetworkLoadBalancer) String() string {
+	return d.TaskName() + "-" + d.Item()
+}
+
+func (e *ClassicLoadBalancer) FindDeletions(c *fi.Context) ([]fi.Deletion, error) {
+	var removals []fi.Deletion
+
+	cloud := c.Cloud.(awsup.AWSCloud)
+
+	for _, nlbName := range e.NLBNamesToDelete {
+		if lb, err := findNetworkLoadBalancerByLoadBalancerName(cloud, nlbName); err != nil {
+			if aerr, ok := err.(awserr.Error); ok && aerr.Code() != elbv2.ErrCodeLoadBalancerNotFoundException {
+				return nil, err
+			}
+		} else if lb != nil {
+			// Deleting a load balancer also deletes its listeners. NLB must be deleted first to delete TG.
```

Comment:
I don't know how deletion tasks are processed: if they're ran in the order that they're returned in the slice and/or if they're retried multiple times until they succeed. I suppose testing will need to confirm this.

Reply:
The order they are processed is the order that they're returned in the slice, and it appears they are retried multiple times until they succeed.